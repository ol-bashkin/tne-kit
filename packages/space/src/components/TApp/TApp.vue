<template>
  <div :class="[classes.app]">
    <div :class="[classes.wrapper]">
      <slot />
    </div>
  </div>
</template>

<script lang="ts" setup>
import { withDefaults, computed } from "vue";

interface Props {
  theme?: string;
}

const props = withDefaults(defineProps<Props>(), {
  theme: "light",
});

const classes = computed(() => {
  return {
    app: `t-app t-app--${props.theme}`,
    wrapper: `t-wrapper primary--base`,
  };
});
</script>
