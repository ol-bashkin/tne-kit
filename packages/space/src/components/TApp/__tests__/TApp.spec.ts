import { mount } from "@vue/test-utils";
import TApp from "../TApp.vue";

describe("TApp.vue", () => {
  test("should have an t-app class", () => {
    const wrapper = mount(TApp, {});
    expect(wrapper.classes()).toContain("t-app");
    expect(wrapper.html()).toMatchSnapshot();
  });
});
