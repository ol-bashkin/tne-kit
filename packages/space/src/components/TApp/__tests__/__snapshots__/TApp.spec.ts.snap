// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TApp.vue should have an t-app class 1`] = `
<div class="t-app t-app--light">
  <div class="t-wrapper primary--base"></div>
</div>
`;
